taxcalc/utils.py

```diff
@@ -231,33 +235,66 @@ def means_and_comparisons(df, col_name, gp, weighted_total):
 
     return diffs
 
+def weighted(df, X):
+    agg = df
+    for colname in X:
+        if colname != 's006':
+            agg[colname] = df[colname]*df['s006']
+    return agg
+
 
 def results(c):
     outputs = [getattr(c, col) for col in STATS_COLUMNS]
     return DataFrame(data=np.column_stack(outputs), columns=STATS_COLUMNS)
 
 
-def create_distribution_table(calc, groupby):
+def weighted_avg_allcols(df, cols):    
+    diff = DataFrame(df.groupby('bins').apply(weighted_mean, "c00100"), columns = ['c00100'])
+    
+    for col in cols:
+        if col == "s006" or col =='num_returns_StandardDed' or col == 'num_returns_ItemDed' or col == 'num_returns_AMT':
+            diff[col] = df.groupby('bins')[col].sum()
+        elif col != "c00100":
+            diff[col] = df.groupby('bins').apply(weighted_mean, col)
+            
+    return diff
+
+def create_distribution_table(calc, groupby, result_type):
     res = results(calc)
+    res['c04470'] = np.where(np.logical_and(res['c00100']>0, res['c04470']>res['_standard']), res['c04470'], 0)
+    res['num_returns_ItemDed'] = np.where(np.logical_and(res['c00100']>0, res['c04470']>0), res['s006'], 0)
+    res['num_returns_StandardDed'] = np.where(np.logical_and(res['c00100']>0, res['_standard']>0), res['s006'], 0)
+    res['num_returns_AMT'] = np.where(res['c09600']>0, res['s006'], 0)
+
+    
     if groupby == "weighted_deciles":
-        gp = groupby_weighted_decile(res)
-    elif groupby == "agi_bins":
-        gp = groupby_income_bins(res)
+        df = groupby_weighted_decile(res)
+    elif groupby == "small_agi_bins":       
+        df = groupby_income_bins(res, "soi")
```

Comment:
There are a few places that "soi" and "tpc" are still popping up. Is this correct?